binding/python/_koala.py

```diff
@@ -0,0 +1,259 @@
+
+        if not os.path.exists(model_path):
+            raise KoalaIOError("Could not find model file at `%s`." % model_path)
+
+        if not os.path.exists(library_path):
+            raise KoalaIOError("Could not find Koala's dynamic library at `%s`." % library_path)
+
+        library = cdll.LoadLibrary(library_path)
+
+        init_func = library.pv_koala_init
+        init_func.argtypes = [c_char_p, c_char_p, POINTER(POINTER(self.CKoala))]
+        init_func.restype = self.PicovoiceStatuses
+
+        self._handle = POINTER(self.CKoala)()
+
+        status = init_func(access_key.encode(), model_path.encode(), byref(self._handle))
+        if status is not self.PicovoiceStatuses.SUCCESS:
+            raise self._PICOVOICE_STATUS_TO_EXCEPTION[status]()
+
+        self._delete_func = library.pv_koala_delete
+        self._delete_func.argtypes = [POINTER(self.CKoala)]
+        self._delete_func.restype = None
+
+        delay_sample_func = library.pv_koala_delay_sample
+        delay_sample_func.argtypes = [POINTER(self.CKoala), POINTER(c_int32)]
+        delay_sample_func.restype = self.PicovoiceStatuses
+        delay_sample = c_int32()
+        status = delay_sample_func(self._handle, delay_sample)
+        if status is not self.PicovoiceStatuses.SUCCESS:
+            self.delete()
+            raise self._PICOVOICE_STATUS_TO_EXCEPTION[status]()
+        self._delay_sample = delay_sample.value
+
+        self._process_func = library.pv_koala_process
+        self._process_func.argtypes = [
+            POINTER(self.CKoala),
+            POINTER(c_short),
+            POINTER(c_short),
+        ]
+        self._process_func.restype = self.PicovoiceStatuses
+
+        self._reset_func = library.pv_koala_reset
+        self._reset_func.argtypes = [POINTER(self.CKoala)]
+        self._reset_func.restype = self.PicovoiceStatuses
+
+        self._sample_rate = library.pv_sample_rate()
+
+        self._frame_length = library.pv_koala_frame_length()
+
+        version_func = library.pv_koala_version
+        version_func.argtypes = []
+        version_func.restype = c_char_p
+        self._version = version_func().decode('utf-8')
+
+        self._pv_free = library.pv_free
+        self._pv_free.argtypes = [c_void_p]
+        self._pv_free.restype = None
+
+    def process(self, pcm: Sequence[int]) -> List[int]:
+        """
+        Processes a frame of audio and returns delayed enhanced audio.
+
+        :param pcm: A frame of audio samples. The number of samples per frame can be attained by calling
+        `.frame_length`. The incoming audio needs to have a sample rate equal to `.sample_rate` and be 16-bit
+        linearly-encoded. Koala operates on single-channel audio. Consecutive calls to `.process()` must provide
+        consecutive frames of audio from the same source, unless `.reset()` has been called in between.
+
+        :return: A frame of enhanced audio samples, stored in a contiguous `numpy` array with `dtype=np.int16`.
```

Comment:
I'm a little out of the loop with the changes but I see the return type is `List[int]`, is the numpy part still applicable?

Reply:
Nope, I just hadn't changed the docstring yet